core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -323,37 +325,17 @@ __wasi_errno_t
 wasmtime_ssp_clock_res_get(__wasi_clockid_t clock_id,
                            __wasi_timestamp_t *resolution)
 {
-#ifdef BH_PLATFORM_WINDOWS
-    return __WASI_ENOSYS;
-#else
-    clockid_t nclock_id;
-    if (!convert_clockid(clock_id, &nclock_id))
-        return __WASI_EINVAL;
-    struct timespec ts;
-    if (clock_getres(nclock_id, &ts) < 0)
-        return convert_errno(errno);
-    *resolution = convert_timespec(&ts);
-    return 0;
-#endif
+    os_clock_res_get(clock_id, *resolution);
+    return convert_errno(errno);
```

Comment:
```suggestion
    if (os_clock_res_get(clock_id, *resolution) != BHT_OK)
        return convert_errno(errno);
    return __WASI_SUCCESS;
```
that way we only convert errno if there was an error, otherwise, we'll return success code. And I think we should do the same for `wasmtime_ssp_clock_time_get`